src/parameterized/normal.jl

```diff
@@ -0,0 +1,166 @@
+using StatsFuns
+export Normal, HalfNormal
+
+# The `@parameterized` macro below does three things:
+# 1. Defines the `Normal` struct 
+#
+#    struct Normal{N, T} <: (ParameterizedMeasure){N}
+#        par::MeasureTheory.NamedTuple{N, T}
+#    end
+#
+#    (Note that this is the same form required by the `@kwstruct` in the
+#    KeywordCalls package.)  
+#
+# 2. Sets the base measure
+#
+#    basemeasure(::Normal) = (1 / sqrt2π) * Lebesgue(ℝ)
+#
+#    This just means that (log-)densities will be defined relative to this.
+#    Including the `(1 / sqrt2π)` in the base measure allows us to avoid
+#    carrying around the normalization constant.
+#   
+# 3. Sets up a method for a call with two unnamed arguments,
+#
+#    Normal(μ, σ) = Normal((μ=μ, σ=σ))
+#
+@parameterized Normal(μ,σ) ≪ (1/sqrt2π) * Lebesgue(ℝ)
+
+@kwstruct Normal(μ,Σ)
+@kwstruct Normal(μ,Σ⁻¹)
+
+# `@kwalias` defines some alias names, giving users flexibility in the names
+# they use. For example, σ² is standard notation for the variance parameter, but
+# it's a lot to type. Some users might prefer to just use `var` and have us do
+# the conversion (at compile time).
+@kwalias Normal [
+    mean  => μ
+    mu    => μ
+    std   => σ
+    sigma => σ
+    var   => σ²
+    cov   => Σ
+]
+
+# It's often useful to be able to map into the parameter space for a given
+# measure. We (currently, at least) do this *independently* per parameter. This
+# allows us to do things like, e.g.
+#
+# julia> transform(asparams(Normal(2,3)))(randn(2))
+# (μ = -0.6778046205440658, σ = 3.7357686957861898)
+#
+# julia> transform(asparams(Normal(μ=-3,logσ=2)))(randn(2))
+# (μ = 0.3995295982002209, logσ = -1.3902312393777492)
+# 
+# Or using types:
+# julia> transform(asparams(Normal{(:μ,:σ²)}))(randn(2))
+# (μ = -0.4548087051528626, σ² = 11.920775478312793)
+#
+# And of course, you can apply `Normal` to any one of the above.
+#
+asparams(::Type{<:Normal}, ::Val{:μ}) = asℝ
+asparams(::Type{<:Normal}, ::Val{:σ}) = asℝ₊
+asparams(::Type{<:Normal}, ::Val{:logσ}) = asℝ
+asparams(::Type{<:Normal}, ::Val{:σ²}) = asℝ₊
+asparams(::Type{<:Normal}, ::Val{:τ}) = asℝ₊
+asparams(::Type{<:Normal}, ::Val{:logτ}) = asℝ
+
+# Rather than try to reimplement everything in Distributions, measures can have
+# a `distproxy` method. This just delegates some methods to the corresponding
+# Distributions.jl methods. For example,
+#
+#     julia> entropy(Normal(2,4))
+#     2.805232894324563
+#
+distproxy(d::Normal{(:μ, :σ)}) = Dists.Normal(d.μ, d.σ)
+
+###############################################################################
+# Some distributions have a "standard" version that takes no parameters
+@kwstruct Normal()
+
+# Instead of setting default values, the `@kwstruct` call above makes a
+# parameter-free instance available. The log-density for this is very efficient.
+logdensity(d::Normal{()} , x) = - x^2 / 2 
```

Comment:
Shouldn't we define the three argument form and have `logdensity(d, x) = logdensity(d, basemeasure(d), x)`

Reply:
I've thought about the tradeoffs here and currently lean toward this approach. But we should discuss this some more to make sure we've thought through it. I'll start a new issue.

Reply:
Resolved here:
https://github.com/cscherrer/MeasureTheory.jl/issues/99